src/contracts/pods/EigenPod.sol

```diff
@@ -50,52 +51,56 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-    uint256 public immutable REQUIRED_BALANCE_WEI;
+    /** 
+    * @notice The value used in our effective restaked balance calculation, to set the 
+    * amount by which to underestimate the validator's effective balance.
+    */
+    uint64 public immutable RESTAKED_BALANCE_OFFSET_GWEI;
 
     /// @notice The owner of this EigenPod
     address public podOwner;
 
     /**
      * @notice The latest block number at which the pod owner withdrew the balance of the pod.
      * @dev This variable is only updated when the `withdraw` function is called, which can only occur before `hasRestaked` is set to true for this pod.
-     * Proofs for this pod are only valid against Beacon Chain state roots corresponding to blocks after the stored `mostRecentWithdrawalBlockNumber`.
+     * Proofs for this pod are only valid against Beacon Chain state roots corresponding to blocks after the stored `mostRecentWithdrawalTimestamp`.
      */
-    uint64 public mostRecentWithdrawalBlockNumber;
+    uint64 public mostRecentWithdrawalTimestamp;
 
     // STORAGE VARIABLES
     /// @notice the amount of execution layer ETH in this contract that is staked in EigenLayer (i.e. withdrawn from the Beacon Chain but not from EigenLayer), 
-    uint64 public restakedExecutionLayerGwei;
+    uint64 public withdrawableRestakedExecutionLayerGwei;
 
     /// @notice an indicator of whether or not the podOwner has ever "fully restaked" by successfully calling `verifyCorrectWithdrawalCredentials`.
     bool public hasRestaked;
 
-    /// @notice this is a mapping of validator indices to a Validator struct containing pertinent info about the validator
-    mapping(uint40 => VALIDATOR_STATUS) public validatorStatus;
+    /// @notice This is a mapping of validatorPubkeyHash to slot to whether or not they have proven a withdrawal for that index
+    mapping(bytes32 => mapping(uint64 => bool)) public provenWithdrawal;
 
-    /// @notice This is a mapping of validatorIndex to withdrawalIndex to whether or not they have proven a withdrawal for that index
-    mapping(uint40 => mapping(uint64 => bool)) public provenPartialWithdrawal;
+    /// @notice This is a mapping that tracks a validator's information by their pubkey hash
+    mapping(bytes32 => ValidatorInfo) internal _validatorPubkeyHashToInfo;
 
     /// @notice Emitted when an ETH validator stakes via this eigenPod
     event EigenPodStaked(bytes pubkey);
 
     /// @notice Emitted when an ETH validator's withdrawal credentials are successfully verified to be pointed to this eigenPod
     event ValidatorRestaked(uint40 validatorIndex);
 
-    /// @notice Emitted when an ETH validator is proven to have a balance less than `REQUIRED_BALANCE_GWEI` in the beacon chain
-    event ValidatorOvercommitted(uint40 validatorIndex);
+    /// @notice Emitted when an ETH validator's  balance is proven to be updated
+    event ValidatorBalanceUpdated(uint40 validatorIndex, uint64 newValidatorBalanceGwei);
```

Comment:
probably want to document what `newValidatorBalanceGwei` is supposed to be. am assuming it's the amount they're being credited with in EigenLayer (as opposed to their effective balance in the Beacon Chain)?

Reply:
added